docs/source/extension/internationalization.rst

```diff
@@ -0,0 +1,106 @@
+Internationalization and Localization
+=====================================
+
+To internationalize your extension, the following tasks are required:
+
+.. note::
+    
+    Please read carefully the :ref:`internationalization-rules` as they are strong constraints for internationalization to work.
+
+1. Add the token ``ITranslator`` from ``@jupyterlab/translation`` package to your plugin dependencies.
+
+.. code:: typescript
+
+    const extension: JupyterFrontEndPlugin<void> = {
+      id: 'jupyterlab-extension',
+      autoStart: true,
+      requires: [ITranslator],
+      activate: (app: JupyterFrontEnd, translator: ITranslator) => {}
+    };
+
+2. Get the translation bundle from the domain on which your extension is translated.
+
+.. code:: typescript
+
+    const trans = translator.load('my_domain');
+
+.. note::
+
+    A good practice is to use your extension named using only letters, numbers and ``_``
+    characters.
+
+    Domain are normalized by replacing ``-`` with ``_`` characters.
+
+3. Wraps all translatable strings with one of the `gettext functions <https://jupyterlab.readthedocs.io/en/latest/api/modules/translation.html#translationbundle>`_.
+
+Examples:
```

Comment:
At some point we should probably add python examples as well, but that can come later, while make the two APIs more uniform? (including interpolation)

Reply:
Yep